src/components/SignUp/SignUp.jsx

```diff
@@ -0,0 +1,104 @@
+import React from "react";
+import css from "./SignUp.module.css";
+import {customHistory} from "../../App";
+
+export class SignUp extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            form: {
+                name: '',
+                email: '',
+                password: '',
+                repeatPassword: ''
+            },
+        }
+    }
+
+    onChange = (event) => {  //Как стрелочная функция заменяет bind();
+        this.setState({
+            form: {
+                ...this.state.form,     //Еще раз про спред оператор
+                [event.target.name]: event.target.value     //Почему ставим квадратные скобки
+            }
+        });
+        console.log('this.state.form: ', this.state.form);
+    };
+
+    onSubmit = (event) => {
+        if(event) {
+            event.preventDefault(); //Что еще интересно можно узнать про методы событий
```

Comment:
https://ru.reactjs.org/docs/events.html

```
boolean bubbles
boolean cancelable
DOMEventTarget currentTarget
boolean defaultPrevented
number eventPhase
boolean isTrusted
DOMEvent nativeEvent
void preventDefault()
boolean isDefaultPrevented()
void stopPropagation()
boolean isPropagationStopped()
DOMEventTarget target
number timeStamp
string type
```

Looks like there are not too many methods there